src/components/Navbar/NavigationDrawer/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import classNames from 'classnames';
+import useTranslation from 'next-translate/useTranslation';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+
+import styles from './ThemeSwitcher.module.scss';
+
+import { themeIcons } from '@/components/Navbar/SettingsDrawer/ThemeSection';
+import Button, { ButtonShape, ButtonSize, ButtonVariant } from '@/dls/Button/Button';
+import PopoverMenu, { PopoverMenuExpandDirection } from '@/dls/PopoverMenu/PopoverMenu';
+import { resetLoadedFontFaces } from '@/redux/slices/QuranReader/font-faces';
+import { selectTheme, setTheme } from '@/redux/slices/theme';
+import ThemeType from '@/redux/types/ThemeType';
+import { logEvent, logValueChange } from '@/utils/eventLogger';
+
+const ThemeSwitcher = () => {
+  const { t } = useTranslation('common');
+  const dispatch = useDispatch();
+  const theme = useSelector(selectTheme, shallowEqual);
+
+  const themes = Object.values(ThemeType).map((themeValue) => ({
+    label: t(`themes.${themeValue}`),
+    value: themeValue,
+  }));
+
+  const onOpenChange = (open: boolean) => {
+    if (open) {
+      logEvent('navigation_drawer_theme_selector_open');
+      return;
+    }
+    logEvent('navigation_drawer_theme_selector_close');
+  };
+
+  const onThemeSelected = (value: ThemeType) => {
```

Comment:
It seems `onSettingsChange` is not being called which is important for logged in users to persist. Please follow `src/components/Navbar/SettingsDrawer/ThemeSection.tsx`'s implementation unless the specs explicitly changes it.